src/lib.rs

```diff
@@ -427,32 +427,37 @@ fn set(
 ///     fail_point!("p1");
 /// }
 ///
-/// mod M {
-///     fn f() {
+/// mod my {
+///     pub fn f() {
 ///         fail_point!("p2");
 ///     }
 /// }
-/// # fn main() {}
+///
+/// # fn main() { f(); my::f() }
 /// ```
-/// The full name of the `p1` fail point is `A::p1`, and `p2` is `A::M::p2`.
+/// The full name of the `p1` fail point is `A::p1`, and `p2` is `A::my::p2`.
 ///
-/// `$e` is used to transform a string to the return type of outer function or closure. If
-/// the return type is `()`, then you can use the `fail_point!($name)` shortcut.
+/// `$e` is used to transform a string to the return type of outer function or closure.
+/// If you don't need to return early or a specified value, then you can use the
+/// `fail_point!($name)`.
 ///
 /// If you provide an additional condition `$cond`, then the condition will be evaluated
 /// before the fail point is actually checked.
 #[macro_export]
 #[cfg(not(feature = "disabled"))]
 macro_rules! fail_point {
+    ($name:expr) => {{
+        let name = concat!(module_path!(), "::", $name);
+        $crate::eval(name, |_| {
+            panic!("Return is not supported for the pattern fail_point!(\"...\")");
```

Comment:
unreachable may be better?

Reply:
I think `panic!`  is fine.

`unreachable!` is 
> A utility macro for indicating unreachable code.
> This is useful any time that the compiler can't determine that some code is unreachable.

For this pattern, actually we can reach here, but it does not support `Return`. Besides, `unreachable!` calls `panic!` internally.